[junnengsoo] iP

junnengsoo opened the pull request:
# Hello ALL! Welcome to the Your Father bot.

This PR addresses what have been included in the bot:
- Todo list
- Deadline list
- Event list

## Implementation Details:

1. I wrote many code
2. It was very difficult..

This is some *good* `code` right here.
```java
// Here's some random code for you to enjoy
public void printMessage(String message) {
     System.out.println(message);
}
```

## Pitstop to check for your well-being
- [ ] Are you happy?
- [ ] Have you showered?
- [ ] Have you had your dinner? 

[Click here](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley) for a nice surprise.
> For real for real 😈

junnengsoo committed:
Complete level 0

junnengsoo committed:
Complete Level-1

junnengsoo committed:
Complete level 2

junnengsoo committed:
Complete Level-3 and add Task class

junnengsoo committed:
Complete Level 4 and add input filter

junnengsoo committed:
Add UITesting

junnengsoo committed:
Complete level-5 by adding exception handling

junnengsoo committed:
Add delet function and modify array to ArrayList

junnengsoo committed:
Add delete function and modify array to ArrayList

junnengsoo committed:
Complete Level-7 and fix delete bug

junnengsoo committed:
Merge pull request #1 from junnengsoo/branch-Level-7

Complete Level-7 and fix delete bug

junnengsoo committed:
Complete level-8 and include more error handling for dateTime

junnengsoo committed:
Merge pull request #2 from junnengsoo/branch-Level-8

Complete level-8 and include more error handling for dateTime

junnengsoo committed:
Complete MoreOOP but unit testing require changes

junnengsoo committed:
Complete MoreOOP but unit testing not modified

junnengsoo committed:
Merge pull request #3 from junnengsoo/branch-A-MoreOOP

Branch a more oop

junnengsoo committed:
Move Parser, Storage and Ui to duke directory

junnengsoo committed:
Merge pull request #4 from junnengsoo/branch-A-Packages

Move Parser, Storage and Ui to duke directory

junnengsoo committed:
Add Gradle

junnengsoo committed:
Merge pull request #5 from junnengsoo/branch-A-Gradle

Add Gradle

junnengsoo committed:
Add JUnit tests for TaskList and Task

junnengsoo committed:
Merge pull request #6 from junnengsoo/branch-A-JUnit

Add JUnit tests for TaskList and Task

junnengsoo committed:
Complete A-Jar

junnengsoo committed:
Merge pull request #7 from junnengsoo/branch-A-Jar

Complete A-Jar

junnengsoo committed:
Modify code styles

junnengsoo committed:
Add JavaDocs for all classes except test classes

junnengsoo committed:
Add find functionality

junnengsoo committed:
Merge pull request #8 from junnengsoo/branch-A-CodingStandard

Modify code styles

junnengsoo committed:
Merge branch 'master' into branch-A-JavaDoc

junnengsoo committed:
Merge pull request #9 from junnengsoo/branch-A-JavaDoc

Add JavaDocs for all classes except test classes

junnengsoo committed:
Merge branch 'master' into branch-Level-9

junnengsoo committed:
Merge pull request #10 from junnengsoo/branch-Level-9

Add find functionality

junnengsoo committed:
Resolve line not uppercase issue

junnengsoo committed:
Merge pull request #11 from junnengsoo/branch-LineIssue

Resolve line not uppercase issue

wasjoe1 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,40 @@
+     * Reads the next line of input from the user.
+     * @return The next line of input from the user.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
```
```suggestion
    public String getCommand() {
        return scanner.nextLine();
    }
```
Since this is a getter method, should this be named as `getCommand`?

junnengsoo replied:
You are absolutely correct, thank you for being an amazing reviewer Joe <3.

wasjoe1 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,40 @@
+        return scanner.nextLine();
+    }
+
+    /**
+     * Prints the welcome message.
+     */
+    public void printMessage(String message) {
+        System.out.println(message);
+    }
+
+    /**
+     * Prints the welcome message.
+     */
```
I liked the naming for these methods! Was wondering if the javadocs for these methods are meant for them?

junnengsoo replied:
It is not, it was gpt generated :)

wasjoe1 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,81 @@
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false; // Initially, the task is not done
```
I like this naming of boolean :)

wasjoe1 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    /**
+     * Returns the description of the task.
+     * @return The description of the task.
+     */
```
Should this formatting for javadocs be changed? (from my recollection, there should be a space between description and return statement)

junnengsoo replied:
Yes, there should be a space, mistake by me

wasjoe1 reviewed (commented):
I like how your code looks generally clean!
Only some nits from my end - the javadocs formatting. Otherwise, LGTM!

junnengsoo committed:
Add GUI interface

junnengsoo committed:
Merge pull request #12 from junnengsoo/branch-Level-10

Add GUI interface

junnengsoo committed:
Add assertion for methods in Duke.java

Add basic assertions to markTaskDone, unmarkTaskDone, findAndListTasks, deleteTask,
addTodo, addDeadline and addEvent.

More assertions can be done for other classes

junnengsoo committed:
Improve code quality

Extract multiple methods from methods in Duke.java

junnengsoo committed:
Merge pull request #13 from junnengsoo/branch-A-Assertions

Add assertion for methods in Duke.java

junnengsoo committed:
Merge conflicts

junnengsoo committed:
Merge pull request #14 from junnengsoo/branch-A-CodeQuality

Branch a code quality

junnengsoo committed:
Add trivia functionality

Users will be able to store trivia questions and answers and test themselves
by calling testtrivia. They will be notified if their answer is correct
or not.

junnengsoo committed:
Merge pull request #15 from junnengsoo/branch-D-Trivia

Add trivia functionality

junnengsoo committed:
Add full commit message

Did not change much in this commit, just trying out hte
full commit message feature.

junnengsoo committed:
Merge pull request #17 from junnengsoo/branch-A-FullCommitMessage

Add full commit message

junnengsoo committed:
Attempt to implement FXML files and Controllers

junnengsoo committed:
Add minor changes

junnengsoo committed:
Refactor JavaFX code to utilise FXML files

junnengsoo committed:
Merge pull request #18 from junnengsoo/branch-JavaFX-Update

Branch java fx update

junnengsoo committed:
Add User Guide

junnengsoo committed:
Add command summary to readme

junnengsoo committed:
Merge pull request #19 from junnengsoo/branch-A-UserGuide

Branch a user guide

junnengsoo committed:
Remove bye command

junnengsoo committed:
Merge pull request #20 from junnengsoo/branch-RemoveBye

Remove bye command

junnengsoo committed:
Modify gradle.build for JAR file release

junnengsoo committed:
Merge pull request #21 from junnengsoo/branch-A-Release

Modify gradle.build for JAR file release
